sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -275,8 +275,13 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
                 .get.withName(FileFormat.ROW_INDEX)
           }
         }
+        // SPARK-41151: metadata column is not nullable for file sources,
+        // [[CreateNamedStruct]] is also not nullable.
```

Comment:
One more thing: This is misleading because this is not true in general.
> [[CreateNamedStruct]] is also not nullable.

To be clear in this context, it would be better to mention `CreateStruct(structColumns)` explicitly instead of saying `[[CreateNamedStruct]] is also not nullable`.

Reply:
Make sense, thanks!